app/src/main/java/com/jerboa/ui/components/settings/Settings.kt

```diff
@@ -0,0 +1,407 @@
+package com.jerboa.ui.components.settings
+
+import android.graphics.Bitmap
+import android.net.Uri
+import android.util.Log
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.jerboa.api.uploadPictrsImage
+import com.jerboa.datatypes.api.SaveUserSettings
+import com.jerboa.db.Account
+import com.jerboa.decodeUriToBitmap
+import com.jerboa.imageInputStreamFromUri
+import com.jerboa.ui.components.common.LargerCircularIcon
+import com.jerboa.ui.components.common.PictrsBannerImage
+import com.jerboa.ui.components.home.SiteViewModel
+import com.jerboa.ui.theme.*
+import kotlinx.coroutines.launch
+
+fun settingsClickWrapper(
+    navController: NavController,
+    account: Account?
+) {
+    account.also {
+        navController.navigate(route = "settings")
+    }
+}
+
+@Composable
+fun SettingsHeader(
+    navController: NavController = rememberNavController(),
+) {
+
+    val backgroundColor = MaterialTheme.colors.primarySurface
+    val contentColor = contentColorFor(backgroundColor)
+
+    TopAppBar(
+        title = {
+            Text(
+                text = "Settings",
+            )
+        },
+        backgroundColor = backgroundColor,
+        contentColor = contentColor,
+        elevation = APP_BAR_ELEVATION,
+        navigationIcon = {
+            IconButton(
+                onClick = {
+                    navController.popBackStack()
+                }
+            ) {
+                Icon(
+                    Icons.Filled.ArrowBack,
+                    contentDescription = "Back"
+                )
+            }
+        },
+    )
+}
+
+@Composable
+fun SettingsField(
```

Comment:
Maybe this should be SettingsTextField. 

Also, Bio is a markdown field, there's a composable for it in common/TextFields.kt .

Reply:
The MarkdownTextField is very specific to CommentReply, let's keep it like this for now and  make it generic some other time.

Reply:
mmk, but add a TODO.